ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/management-security.j2

```diff
@@ -63,6 +63,32 @@ management security
 {%             endif %}
 {%         endfor %}
 {%     endif %}
+{%     for profile in management_security.shared_secret_profiles | arista.avd.natural_sort("profile") %}
+   session shared-secret profile {{ profile.profile }}
+{%         for secret in profile.secrets | arista.avd.natural_sort("name") %}
+{%             if secret.secret is arista.avd.defined %}
+{%                 set secret_cli = "secret " ~ secret.name ~ " " ~ secret.secret_type | arista.avd.default("7") ~ " " ~ secret.secret | arista.avd.hide_passwords(hide_passwords) %}
+{%                 if secret.receive_lifetime.infinite is arista.avd.defined(true) %}
+{%                     if secret.transmit_lifetime.start_date_time is arista.avd.defined and secret.transmit_lifetime.end_date_time is arista.avd.defined %}
+{%                         set secret_cli = secret_cli ~ " receive-lifetime infinite transmit-lifetime " ~ secret.transmit_lifetime.start_date_time ~ " " ~ secret.transmit_lifetime.end_date_time %}
+{%                     elif secret.transmit_lifetime.infinite is arista.avd.defined(true) %}
+{%                         set secret_cli = secret_cli ~ " receive-lifetime infinite transmit-lifetime infinite" %}
+{%                     endif %}
+{%                 elif secret.receive_lifetime.start_date_time is arista.avd.defined and secret.receive_lifetime.end_date_time is arista.avd.defined %}
+{%                     set secret_cli = secret_cli ~ " receive-lifetime " ~ secret.receive_lifetime.start_date_time ~ " " ~ secret.receive_lifetime.end_date_time %}
+{%                     if secret.transmit_lifetime.start_date_time is arista.avd.defined and secret.transmit_lifetime.end_date_time is arista.avd.defined %}
+{%                         set secret_cli = secret_cli ~ " transmit-lifetime " ~ secret.transmit_lifetime.start_date_time ~ " " ~ secret.transmit_lifetime.end_date_time %}
+{%                     elif secret.transmit_lifetime.infinite is arista.avd.defined(true) %}
+{%                         set secret_cli = secret_cli ~ " transmit-lifetime infinite" %}
+{%                     endif %}
+{%                 endif %}
+{%                 if secret.local_time is arista.avd.defined(true) %}
+{%                     set secret_cli = secret_cli ~ " local-time" %}
+{%                 endif %}
+      {{ secret_cli }}
```

Comment:
You could end up generating bad config since you don't check at the beginning for all "legal" combinations. Imagine not having receive_lifetime set at all, you would fall through the if and elif and maybe then add local-time.
I suggest chaing the `if` in line 69 to instead check for all the required keys like `secret.name and secret.secret and (receive_lifetime.infinite or (receive_lifetime.start_date_time and receive_lifetime.end_date_time)) and (transmit_lifetime.infinite or (transmit_lifetime.start_date_time and transmit_lifetime.end_date_time))`.
This will also allow you to optimize the logic below like I requested earlier, so you can just check for the receive setting in isolation and add to cli var and then the same for transmit.

Reply:
Fixed.

Reply:
You forgot the last part. The logic below the new large if, can be simplified because you know that the variables are there.

Reply:
I have submitted a new suggestion for improving the logic so I will close this one.